src/roller/table.ts

```diff
@@ -38,40 +43,148 @@ export class TableRoller extends GenericFileRoller<string> {
     async build() {
         this.resultEl.empty();
         const result = [this.result];
-
         if (this.plugin.data.displayResultsInline) {
             result.unshift(this.inlineText);
         }
-
         MarkdownRenderer.renderMarkdown(
             result.join(""),
             this.resultEl.createSpan("embedded-table-result"),
             this.source,
             null
         );
     }
+
+    // Quick and Dirty
+    // TODO: to be refactored (using regex maybe?)
+    prettify(input: string): string {
+        let tab = "\t";
+
+        let tabCount = 0;
+        let output:string = "";
+
+        for (let i = 0; i < input.length; i++) {
+            if (input.charAt(i) == "(") {
+                tabCount++;
+                output += "(\n";
+                output += tab.repeat(tabCount);
+            }
+            else if (input.charAt(i) == ")") {
+                tabCount--;
+                output += "\n";
+                output += tab.repeat(tabCount);
+                output += ")";
+            }
+            else if (input.charAt(i) == ";") {
+                output += ",\n";
+                output += tab.repeat(tabCount);
+            }
+            else if (input.charAt(i) == "|" && input.charAt(i-1) == "|") {
+                output += "|\n";
+                output += tab.repeat(tabCount);
+            }
+            else {
+                output += input.charAt(i);
+            }
+        }
+        return output;
+    }
+
+    async getSubResult(input: string): Promise<SubRollerResult> {
```

Comment:
Below you say that input might be a number. I would suggest making that explicit in the typing.

Reply:
Another one I wanted to discuss (I'm not familiar enough with typescript).

I have been surprised to realize that: 

```
function f(input: string) {
    console.log(typeof input);
}
const n:number = 12;
f(n);
```
is considered valid and ... displays "number" in the console!

In fact I **don't** want to allow calling getSubResult with a number, just with a string.

If your input is a number, then convert it to a string before calling me.
I thought that **input:string** would "guarantee" that but... that's not the case (I can call it with any type apparently).
(yeah, I know ... typescript is not C++ :-p)

Not really sure what to do in fact!

Reply:
Yep... typescript is just syntactic sugar. The types have no bearing on what is actually generated in the code (outside of `enums` but that's a separate thing).

All you're doing is telling the typescript language service that the input is a string. In practice you can call the method with anything... the idea is that by telling the language service "this is only ever a string", you will receive an error message from the language service if you call the method with a parameter that is _not_ a string, and you can take it a step further and make the build fail on these types of errors. _But once the JavaScript is generated, it doesn't matter anymore._

Basically my suggestion is to say `input: string | number` and then the language service will show you if there is a possibility that passing in a number will throw an error.

Reply:
OK, I know what I wasn't understanding.

getSubResult is called with subRoller.result which is a BasicRoller where result is... `any` and as soon as there's an any, all type checking are "ignored".

Anyway, I'll do as you suggested (even if that hurts my brain :-p)